zproject/backends.py

```diff
@@ -302,6 +302,9 @@ class EmailAuthBackend(ZulipAuthMixin):
     Allows a user to sign in using an email/password pair.
     """
 
+    auth_backend_name = 'Email'
```

Comment:
This commit besides settings the attrs on these backends, should probably also move the "undeclared" once into `ZulipAuthMixin`from `ExternalAuthMethod`. And could we use this to clean up what Tim posted
```
AUTH_BACKEND_NAME_MAP: Dict[str, Any] = {
    'Dev': DevAuthBackend,
    'Email': EmailAuthBackend,
    'LDAP': ZulipLDAPAuthBackend,
}
```
somehow?

Reply:
Okay, it's probably better to move the "undeclared" ones to `ExternalAuthMethod`.
(just to mention `ZulipRemoteUserBackend` subclasses `ExternalAuthMethod` and do not subclass `ZulipAuthMixin`, and it has those attributes we're moving, I do not think this would be problem, but if we want to add a logger to it later we might want to so `logger = logging.getLogger('f'zulip.auth.{name}')` to it. I'm confused why we're not subclassing it to `SocialAuthMixin` :thinking:)


> And could we use this to clean up what Tim posted

I could think of two ways but I feel like that is simpler/ better than what we could do. 
we could use some decorator like `external_auth_method` that'd add them to `AUTH_BACKEND_NAME_MAP` or we could try [getting subclasses of ZulipAuthMixin](https://stackoverflow.com/questions/3862310/how-to-find-all-the-subclasses-of-a-class-given-its-name) to create the `AUTH_BACKEND_NAME_MAP` and remove irrelevant ones like `SocialAuthMixin`.

Reply:
Force pushed the last commit to move the two attrs with "undeclared" to `ZulipAuthMixin`.